### PR TITLE
Handle create posts with images files 

### PR DESCRIPTION
**Pull request by chaitali**
Added handling for creating posts and images

### CODE REVIEW

File: core/posts/posts_view.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    # if image, make another post with generated post_id
+    # TODO Multi Image Posts?
+    if (request.FILES):
+        new_id = new_post.post_id
+        image_type = request.FILES["imageFiles"].content_type
+        
+        if image_type == "image/jpeg" or image_type == "image/png":
+            content_type = f"{image_type};base64"
+        else:
```
Comment (amalik2) [inline core/posts/posts_view.py]: Shouldn't there be some kind of file size limit on the image?

File: core/urls.py [hunk 2]
```diff
@@ -18,13 +18,15 @@
...
 from core.authors.friend_request_view import handle_follow_request as friend_request
+from core.posts.posts_view import handle_posts as handle_posts
 from core.authors.login_view import login
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('users/', include('core.users.urls')),
     path('author/', include('core.authors.urls')),
     path('posts/', include('core.posts.urls')),
+    path('handleposts/', handle_posts, name='handleposts'),
```
Comment (amalik2) [inline core/urls.py]: I think the [path to make a new post](https://github.com/abramhindle/CMPUT404-project-socialdistribution/blob/master/project.org#requirements) is /posts/POST_ID, but [this](https://github.com/abramhindle/CMPUT404-project-socialdistribution/blob/master/example-article.json#L259) says that path is meant for querying for a specific post. You might want to ask for clarification on that on the forums

Regardless, we still need a PUT method on /posts/POST_ID

File: core/posts/posts_view.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        serializer = PostsSerializer(data=data)
+        serializer.is_valid()
+        serializer.save()
+
+    return success, message
+
+
+@api_view(['POST'])
+def handle_posts(request):
```
Comment (amalik2) [inline core/posts/posts_view.py]: Wouldn't it make more sense to have a separate path and view for deleting a post? Using DELETE for deleting a post would probably be better as well

File: core/posts/posts_view.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        image_type = request.FILES["imageFiles"].content_type
+        
+        if image_type == "image/jpeg" or image_type == "image/png":
+            content_type = f"{image_type};base64"
+        else:
+            success = False
+            message = "Invalid file type"
+            return success, message
+
+        # https://stackoverflow.com/questions/44489375/django-have-admin-take-image-file-but-store-it-as-a-base64-string
```
Comment (MandyMeindersma) [inline core/posts/posts_view.py]: add the username here, hindle will be picky about that

File: core/posts/posts_view.py (hunk 1 above)
Comment (cmackenzie1) [reply core/posts/posts_view.py]: I would suggest enforcing a limit on both the frontend and backend.

Comment (chaitali) [reply core/posts/posts_view.py]: Added a 10MB backend limit

File: core/posts/posts_view.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            image_type = img_file.content_type
+            image_size = img_file.size
+            
+            if image_type == "image/jpeg" or image_type == "image/png":
+                content_type = f"{image_type};base64"
+            else:
+                return False, "Invalid file type", None
+
+            # http://www.learningaboutelectronics.com/Articles/How-to-restrict-the-size-of-file-uploads-with-Python-in-Django.php
+            if image_size > 10485760:
```
Comment (cmackenzie1) [inline core/posts/posts_view.py]: Move to constant `MAX_UPLOAD_SIZE`

**Pull request merged**